### PR TITLE
⬆️ Upgrade .NET to 7.0

### PR DESCRIPTION
**Pull request by ostorc**

### CODE REVIEW

**Comment by stepech** [comment]
On hold until there is a bit wider support for dotnet-7

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "6.0.100",
+    "version": "7.0.*",
```
Comment (stepech) [inline global.json]: This is going to fail with CodeQL, as it again doesn't yet fully support dotnet 7. It is better to add explicit version now and fall back to wildcards later when they add it.
```suggestion
    "version": "7.0.100",
```

**Comment by ostorc** [comment]
image

**Comment by ostorc** [comment]
- Fedora 35: .NET Core 3.1 and .NET 6 are currently available
- Fedora 36: .NET Core 3.1 and .NET 6 are currently available
- Fedora 37 (about to be released in a few days): Only.NET 6 is currently available
- .NET 7 packages are being reviewed for inclusion in Fedora 36 and later. Fedora 35 is being left out because it goes EOL in about a month or so.

**Review by stepech — APPROVED** [review]

**Pull request merged**